### DISCUSSION TITLE
Different form for create vs edit

### DISCUSSION THREAD

**Discussion by jonocodes** [aminalaee/sqladmin#692] [Q&A] [answered]
Some fields I would like to not be changeable after the object is created.
So on the edit form those fields would be disabled.

**Comment by johanboekhoven** [comment]
https://aminalaee.dev/sqladmin/configurations/
Form options should be what you are looking for, not sure if you can easily display a field without being able to edit them, but you can exclude it seems.

**Reply by jonocodes** [reply]
I'll clarify. The form options are for both the 'create' and 'update' views.
I would like to show 3 fields in the 'create' view, and 1 field in the 'update' view.

**Reply by wnowicki** [reply, marked as answer]
I think this is now solved

This works for creation and edit:

```python
form_columns = [
    "field_one",
    "field_two",
    "field_three",
]
```

but later you can add:

```python
form_edit_rules = [
    "field_two",
]
```

and this will limit edit form only to `field_two`, please see https://aminalaee.github.io/sqladmin/configurations/#form-options

**Comment by jonocodes** [comment]
Here is my half working hacky solution that basically monkey patches in some additional attributes only when the 'edit' form is being rendered. It then undoes it, so future renders are not affected.

```python

class UsersAdmin(ModelView, model=Users):
	...
	
    form_columns = ['email', 'organization', 'name']
    
    form_widget_args_update = dict(email=dict(readonly=True))
	...


class MyAdmin(Admin):

	...
	
    @login_required
    async def edit(self, request: Request) -> Response:

        identity = request.path_params["identity"]
        model_view = self._find_model_view(identity)

        # monkey patch the 'edit' specific form properties
        if model_view.form_widget_args_update:
            orig = model_view.form_widget_args
            model_view.form_widget_args = model_view.form_widget_args_update

        make_edit = await super().edit(request)

        # undo change
        if model_view.form_widget_args_update:
            model_view.form_widget_args = orig

        return make_edit


```
